File: C++ ad y ad/3_ Orientado a objetos/S05.1_ La observacion/Solucion_la_observacion/La_observacion/src/piedra/piedra.cpp
#include "piedra.h"

#include <iostream>


Piedra::Piedra(string tamanio, string color)
{
	this->tamanio = tamanio;
	this->color = color;
}
Piedra::~Piedra()
{
	cout << "La piedra " << tamanio << " de color " << color << " fue liberada de la memoria." << endl << endl;
}


string Piedra::getTamanio()
{
	return tamanio;
}
string Piedra::getColor()
{
	return color;
}

File: C# ad y ad/2_ Orientado a objetos/S05.1_ La observación/Solucion_la_observacion/La_observacion/Src/Piedra/Piedra.cs

namespace La_observacion
{
	class Piedra
	{
		string tamanio;
		string color;


		public Piedra(string tamanio, string color)
		{
			this.tamanio = tamanio;
			this.color = color;
		}


		public string GetTamanio()
		{
			return tamanio;
		}
		public string GetColor()
		{
			return color;
		}
	}
}
